File: C++/Medium/RuntimeError/smallest-integer-divisible-by-k_7.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        
        int n = 1;
        int len = 1;
        while (n % k != 0) {
            n = n * 10 + 1;
            len++;
            if (len > k) return -1;
        }
        return len;
    }
};

File: Java/Medium/Accepted/smallest-integer-divisible-by-k_7.java
class Solution {
    public int smallestRepunitDivByK(int k) {
        
        if (k % 2 == 0 || k % 5 == 0) return -1;
        int n = 1;
        int mod = 0;
        while (n <= k) {
            mod = (mod * 10 + 1) % k;
            if (mod == 0) return n;
            n++;
        }
        return -1;
    }
}